### ISSUE TITLE
JITServer AOT test

### ISSUE THREAD

**Issue by llxia** [eclipse-openj9/openj9#16428]
Thanks @mpirvu's suggestion. We are going to add a JITServer AOT test by leveraging the existing system test - [MiniMix_5m](https://github.com/adoptium/aqa-tests/blob/ac712db0d0aa78b18cc7a88332479a31baf5f637/system/otherLoadTest/playlist.xml#L35). We will mimic the scenario of the server and client, but to make the testing easy, we will run everything on one machine. We may also want to increase `MiniMix_5m` to 10mins (e.g., `MiniMix_10m`). 

Details:
- Start JITServer with -XX:+JITServerUseAOTCache
- Run `MiniMix_10m` with -Xshareclasses and -XX:+JITServerUseAOTCache
- Destroy the sharedclasses at the client
- Run `MiniMix_10m` again with -Xshareclasses and -XX:+JITServerUseAOTCache
- Stop JITServer 
- clean up

Related: https://github.com/eclipse-openj9/openj9/issues/15401

**Comment by llxia** [comment]
@LongyuZhang could you take care of this? Thanks

**Comment by cjjdespres** [comment]
This would be good to have as a start on adding more JITServer AOT cache tests to the test suite, especially since we've been making some fairly significant changes to the AOT cache recently. Could work on this be restarted?

**Comment by llxia** [comment]
@cjjdespres Do we still need the MiniMix_10m with JITServer with -XX:+JITServerUseAOTCache test that is mentioned above? Also, can you provide more details about `adding more JITServer AOT cache tests to the test suite`? What tests/scenarios are you referring to?

**Comment by cjjdespres** [comment]
Yes, I I think the test outlined in the https://github.com/eclipse-openj9/openj9/issues/16428#issue-1482226037 is needed.

I don't think we have a list of exactly the extra tests/scenarios that should be added. That's just a goal of ours, to improve JITServer AOT cache testing coverage.

One other scenario I can think of is the test in https://github.com/eclipse-openj9/openj9/issues/16428#issue-1482226037, except that the client running MiniMix_10m will have the options `-XX:+JITServerUseAOTCache -XX:+JITServerAOTCacheIgnoreLocalSCC` in both runs. The client might or might not have `-Xshareclasses` set up in both cases, I suppose.

Maybe @mpirvu and @AlexeyKhrabrov have ideas.

**Comment by mpirvu** [comment]
I am thinking of making `-XX:+JITServerAOTCacheIgnoreLocalSCC` the default, so going forward the sequence mentioned in the description (https://github.com/eclipse-openj9/openj9/issues/16428#issue-1482226037) is good enough.
A variation could be where the client does not specify `-Xshareclasses`, but only `-XX:+JITServerUseAOTCache`

Further down the line we may want to test AOT cache persistence. The sequence of operations in that case is:
1. Start JITServer with `-XX:+JITServerUseAOTCache -XX:+JITServerAOTCachePersistence`
2. Start client with  `-XX:+JITServerUseAOTCache` 
3. Remove client, remove server
4. Repeat step 1
5. Repeat step 2

**Comment by llxia** [comment]
https://github.com/eclipse-openj9/openj9/issues/16428#issue-1482226037 and https://github.com/eclipse-openj9/openj9/issues/16428#issue-1482226037 describe single tests. 

Do we have a plan to run all tests with JITServer with `-XX:+JITServerUseAOTCache -XX:+JITServerAOTCacheIgnoreLocalSCC` or `-XX:+JITServerUseAOTCache -XX:+JITServerAOTCachePersistence`?

We currently have TEST_FLAG= JITAAS, where JITServer will start before running tests
https://github.com/adoptium/TKG/blob/6e73a9fa473738b6e92c7a65e59588b5b0f408a8/testEnv.mk#L23-L26

Is this something that we will do for JITServer AOT?

**Comment by cjjdespres** [comment]
> Is this something that we will do for JITServer AOT?

I think that could be valuable for `-XX:+JITServerUseAOTCache -XX:+JITServerAOTCacheIgnoreLocalSCC`. I caught a couple of bugs doing that myself manually, running `sanity.functional` with a JITServer with AOT cache enabled running on the side throughout.

Would the tests be run twice in this mode? That would hopefully test both the compiling/storing of methods and the loading of methods in the JITServer AOT cache.

The `-XX:+JITServerAOTCachePersistence` feature works by periodically saving the content of the AOT cache to a file on disk, so that future JITServer instances that start up can read the cache from that file. We could test that in the same way as `TEST_FLAG=JITAAS`, but we'd have to make sure that fresh servers are started somewhat regularly so the reading-from-disk part of the feature was actually tested.

**Comment by cjjdespres** [comment]
Now that https://github.com/eclipse-openj9/openj9/pull/19584 has been merged, the JITServer no longer needs the option `-XX:+JITServerUseAOTCache`. The client will still need the option.
